src/api/store/utils/hashing.ts

```diff
@@ -0,0 +1,19 @@
+import { hexlify } from '@ethersproject/bytes';
+import { sign, utils, verify } from 'noble-ed25519';
+
+import type { JsonRPCRequest } from '@types';
+import { stripHexPrefix } from '@utils';
+
+export const hashRequest = async (data: JsonRPCRequest) => {
+  const encoded = Buffer.from(JSON.stringify(data), 'utf-8');
+  const buffer = await utils.sha512(encoded);
+  return stripHexPrefix(hexlify(buffer));
+};
+
+export const signRequest = async (hash: string, privateKey: string) => {
+  return sign(hash, privateKey);
+};
+
+export const verifyRequest = async (sig: string, hash: string, publicKey: string) => {
```

Comment:
Maybe pass in the entire request object instead of the hash, and handle hashing in this function, rather than in the request handler.